pkg/reconciler/contour/contour.go

```diff
@@ -332,3 +333,11 @@ func (r *Reconciler) lbStatus(ctx context.Context, vis v1alpha1.IngressVisibilit
 	}
 	return
 }
+
+func isDomainMapping(ing *v1alpha1.Ingress) bool {
+	_, hasDmUIDLabelKey := ing.Labels[serving.DomainMappingUIDLabelKey]
```

Comment:
I had to bring in a fair amount of serving code just to use these labels. Maybe there is a better way to import it that doesn't bring in so much?

Reply:
What you probably want to do is to either copy the constants here, or move them into `networking`.  When you have `net-contour` import `serving` directly, it makes it hard to cut a release, because the `net-contour` package can't be cut until the `serving` package is tagged.

Alternatively, you could look at the `OwnerReferences` field to see whether the owner is a Knative Route or a DomainMapping.

The third option would be to add some explicit field to the KIngress indicating which type it is.

The fourth option would be to look at the `spec.rules` -- if there's a single `rule` with `visibility: externalIP`, that must be a DomainMapping, because a Route always has an internal mapping: https://github.com/knative/serving/blob/main/pkg/reconciler/route/resources/ingress.go#L141

Reply:
ah okay, thanks for explaining that about importing serving. 

I didn't even think about the owner reference, I think I will go for that. It seems the least intrusive but still pretty explicit. Then I can avoid having duplicated constants or adding more fields to the kingress.

Reply:
DomainMapping just sets `HostRewrite` on the KIngress - https://github.com/knative/networking/blob/2473e65d69206ed544f32e47f994cf3635bebdbe/pkg/apis/networking/v1alpha1/ingress_types.go#L208-L212

You could look at that spec property and the target k8s service (it pointing to a contour's envoy service)

Reply:
ah, well maybe not, the type definition for domainmappings is in the serving repo.

Reply:
I think you can define a constant for DomainMapping here, or add a helper in `networking` to determine if a KIngress represents a DomainMapping.

It might be more robust, though, to look for the `rewriteHeader` field and the presence of `knative.dev/networking/pkg/http/header.OriginalHostKey` as a field in `AppendHeaders`.